tests/users/test_models.py

```diff
@@ -1122,6 +1122,21 @@ def test_latest_common_approval_when_only_pe_approval(self):
         pe_approval = PoleEmploiApprovalFactory(nir=user.jobseeker_profile.nir)
         assert user.latest_common_approval == pe_approval
 
+    def test_latest_common_approval_with_expired_pe_approval(self):
+        user = JobSeekerFactory()
+        today = timezone.localdate()
+        PoleEmploiApprovalFactory(
+            nir=user.jobseeker_profile.nir,
+            start_at=today - datetime.timedelta(days=30),
+            end_at=today - datetime.timedelta(days=1),
+        )
+        approval = ApprovalFactory(
+            user=user,
+            start_at=today - datetime.timedelta(days=10),
+            end_at=today - datetime.timedelta(days=7),
```

Comment:
```suggestion
            expired=True,
```

Reply:
Non. Il ne faut pas juste que le PASS ait expiré, il faut que le PASS ait expiré AVANT l’agrément PE. Mettons les dates en fonction.

Reply:
Euh, t'es sûr ? :thinking:
On vérifie (et retourne) les PASS IAE en priorité mais si celui-ci est expiré et en période de carence alors on passe sur le dernier agrément, on compare pas les dates entres les deux de ce que je vois.

Reply:
Noui. J’ai répondu de mémoire et ma réponse est inexacte : il faut que `self.latest_approval.is_in_waiting_period` (donc fin du pass il y a moins de 2 ans, ce qui n’est pas le cas de `expired=True`) et que `not self.latest_pe_approval.is_valid()` [1] donc `PoleEmploiApproval.end_at < now`, mais sans contrainte entre les fins du pass et de l’agrément.

[1] pour le test, où on ne veut pas rentrer dans
```python
        if (
            self.latest_approval
            and self.latest_approval.is_in_waiting_period
            and self.latest_pe_approval
            and self.latest_pe_approval.is_valid()
        )
            return self.latest_pe_approval
```

Reply:
_relis le commentaire du trait : "# Use old (but realistic) dates so `expired` can be used anywhere without triggering specials cases"_
Je devais avoir plus de neurones connectés le jour où j'ai fait ça, tant mieux j'ai envie de dire ! :grin: